RecoParticleFlow/PFClusterProducer/plugins/PFHBHERecHitCreatorCUDA.cu

```diff
@@ -537,26 +483,16 @@ namespace PFRecHit {
     void entryPoint(::hcal::RecHitCollection<::calo::common::DevStoragePolicy> const& HBHERecHits_asInput,
 		    const PFRecHit::HCAL::Constants& cudaConstants,
 		    const ConstantProducts& constantProducts,
-		    //const PFRecHit::HCAL::ConstantProducts& constantProducts,
                     OutputPFRecHitDataGPU& HBHEPFRecHits_asOutput,
                     PersistentDataGPU& persistentDataGPU,
                     ScratchDataGPU& scratchDataGPU,
                     cudaStream_t cudaStream,
                     std::array<float, 5>& timer) {
 
-      //printf("bb %8d\n",recHitParametersProduct.valuesdepthHB[1]);
-      //printf("bb %8d\n",constantProducts.depthHB[1]);
       std::cout << constantProducts.denseId.size() << std::endl;
```

Comment:
Put these lines in if (debug) {}.
It may help you to see their sizes printed out, so you see how their sizes are different and why...
denseId takes values of 0 .... 2xxxx, almost contiguous but not quite.
This array has a sizes of about 15xxx or so.

Then, detId and position arrays have sizes of 2xxxx, and indexed by denseid.
Good thing of denseId is that, by a rule, we can covert detId to denseId, which is close to contiguous, and suitable for e.g. indexing.
neighbhours arrays has size of 2xxxx * 8, because we consider 8 neighbors.